### ISSUE TITLE
Can't find app in store?

### ISSUE THREAD

**Issue by james-cook** [matiasdelellis/facerecognition#214]
Hallo everyone,

I installed facerecognition last summer and had quite a few adventures with at that time :)
Now I've got the whole thing in docker on a brand new host droplet (Docker Alpine Nextcloud 17.xx)

**Nextcloud is up and running.**

I installed pdlib into the docker as shown in https://github.com/matiasdelellis/facerecognition/issues/160#issuecomment-559590851

> FROM nextcloud:fpm-alpine
> 
> RUN apk --update add --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing php7-pdlib
 (repo spec may not be necessary anymore - but this still works)

And now I should just need to find face recognition App and install it...
https://github.com/matiasdelellis/facerecognition/issues/1#issuecomment-558953708 says:

> For people who are using the Alpine based Docker image of Nextcloud, installation would involve just running apk add php7-pdlib in the container/Dockerfile, instead of building the whole thing.

And from: https://github.com/matiasdelellis/facerecognition/wiki/Installation

> Install FaceRecognition app
> In general, there is no need to install FaceRecognition from source, please use Nextcloud app store -
> 

It's late at night and I'm probably missing something obvious - but I don't see the app in **my** nextcloud instance under /apps/multimedia (with a hoped for install/enable button :) )

**Comment by matiasdelellis** [comment]
Hi @james-cook 

Glad to see you here again ... :smiley: 

>  I installed pdlib into the docker as shown in #160 (comment)

Wow.. Great that they published the packages .. :smile: 

> It's late at night and I'm probably missing something obvious - but I don't see the app in my nextcloud instance under /apps

Now it is published as nigthly but nextcloud does not publish these versions. See these comment:

> https://github.com/matiasdelellis/facerecognition/issues/1#issuecomment-557033287

Soon I will publish it more officially but in meantime continue with the manual steps .. Sorry..

**Comment by james-cook** [comment]
Ah, ok... l thought it might be something like this...
Looks like I jumped the gun :( AFAICT it turns out the pdlib was not installed after all using:

> FROM nextcloud:fpm-alpine
> RUN apk --update add --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing php7-pdlib
I don't get an error(?) but doing a `docker exec --user root nextcloud php --info `
I DO NOT see: 

> pdlib
> 
> pdlib support => enabled
> 

(I also tried: 
> RUN apk --update add php7-pdlib
)

In  nextcloud the local facerecognition app page says:

> This app cannot be installed because the following dependencies are not fulfilled:
> The library pdlib is not available.

So, I'm wondering whether I'm missing a step - https://github.com/matiasdelellis/facerecognition/issues/160 seems to suggest this should be all I need

**Comment by james-cook** [comment]
My bad,  I see actually that https://github.com/matiasdelellis/facerecognition/issues/160#issuecomment-559590851
does say:

> Once alpinelinux/aports#12174 has been merged,

I had assumed since that issue is closed that the merge had taken place...

**Comment by james-cook** [comment]
Building the dlib dependencies in the Dockerfile looks to be working for alpine linux.

**Comment by james-cook** [comment]
I'll close this now. 
I have everything running.

- I use a dockerfile to compile/install the dlib dependencies.
- I installed the facerecognition app by hand, using the git clone... etc. instructions.
- It's important to `chown -R www-data:root facerecognition/` before enabling the app in the browser (in fact you can't until this is done).
- Then in my docker based setup with www-data not having their own shell the occ command becomes:
`su www-data -s /bin/sh -c "php occ face:background_job"`

**Comment by matiasdelellis** [comment]
> I'll close this now.
> I have everything running.

Wow.. Great!.

Please @james-cook 
Try not to comment in closed reports due it is difficult to follow .. :sweat_smile: 
In any case first reopen it and then comment that there is any problem with it .. :wink:

**Comment by agboom** [comment]
Hi @james-cook, did you get it working with the Alpine package ` php7-pdlib` or even just `dlib`? Or did you need to build one or both packages?

I contributed those packages to Alpine, but my experience is still somewhat limited w.r.t. building and packaging C libraries. As far as I can see I'm getting some linking or shared library error when starting Nextcloud from Docker.

The error is:
```
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib/php7/modules/pdlib.so' (tried: /usr/lib/php7/modules/pdlib.so (Error relocating /usr/lib/php7/modules/pdlib.so: dgesdd_: symbol not found), /usr/local/lib/php/extensions/no-debug-non-zts-20180731//usr/lib/php7/modules/pdlib.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20180731//usr/lib/php7/modules/pdlib.so.so: No such file or directory)) in Unknown on line 0
```

The Dockerfile is:
```
FROM nextcloud:17-fpm-alpine

ADD repositories /etc/apk

RUN apk --update upgrade

RUN apk add --no-cache php7-pdlib
RUN echo 'extension=/usr/lib/php7/modules/pdlib.so' > /usr/local/etc/php/conf.d/pdlib.ini
RUN echo 'memory_limit=2048M' > /usr/local/etc/php/conf.d/memory-limit.ini
```

Where `repositories` is:
```
http://dl-cdn.alpinelinux.org/alpine/edge/main
http://dl-cdn.alpinelinux.org/alpine/edge/community
http://dl-cdn.alpinelinux.org/alpine/edge/testing
```

Do you have a similar experience or different?

> Try not to comment in closed reports due it is difficult to follow ..

Sorry for not honoring your rule here, but this issue might need to be reopened if the Alpine package doesn't work, right?

**Comment by james-cook** [comment]
Yes, I got **exactly** that error message :) 

The way I read the error message (and I could be wrong!) is that dlib "needs" or "is looking for" or "relies" on the symbol `dgesdd_` being in `pdlib.so` but not finding it there, so can't load the library. 


So currently I compile/make etc. both dlib and pdlib .- using pretty much the code here: https://github.com/matiasdelellis/facerecognition/issues/175#issuecomment-554747326
(I saw your Dockerfile a comment or so later on - WITH the install of facerecognition - but I haven't tried that yet) (Note: I need another install run just to verify this).

Previously I had used a Dockerfile from elsewhere (the dlib/pdlib project?) but was getting strange side effects in nextcloud itself (https://github.com/matiasdelellis/facerecognition/issues/216). 

The backstory is I installed facerecognition plus many other things on a "production" nextcloud in the summer 2019 - and had lots of underlying problems - including in the logs  - "JPEG parameter struct mismatch: library thinks size is 656, caller expects 624" 
(see your ticket: https://github.com/matiasdelellis/facerecognition/issues/175 and my ticket at the time https://github.com/matiasdelellis/facerecognition/issues/149#issuecomment-503199719).

Everything was VERY fragile so I decided to Dockerize to try and get to a stable installation.

So far IMHO it appears these kinds of errors are coming from how dlib/pdlib.so are compiled (which libs are used/shared) and installed...
